TypeInjections/Tool/Tool.fs

```diff
@@ -21,77 +57,69 @@ module Tool =
         let commitRepo = newRepo.Lookup<Commit>(hash)
         Commands.Checkout(newRepo, commitRepo) |> ignore
 
-    // based off command line flags, create and provide path for input/output folders
-    let interpretFlag (argvList: string list) =
-        let flag = argvList.Item(0)
+    let runGit (git: GitOptions) : unit =
+        Utils.log "***** git commit"
+        // location of repo
+        let location = Directory.GetCurrentDirectory()
 
-        // -p [inputPath] [inputPath] [outputPath]
-        if flag.Contains("-p") then
-            Console.WriteLine("***** local project comparison")
+        let commit1, commit2 = git.oldHash, git.newHash
+        let path = git.outputFolder
+        let pathOld = $"{path}/Old"
+        let pathNew = $"{path}/New"
 
-            if argvList.Length <> 4 then
-                failwith "usage: program -p OLD NEW OUTPUTFOLDER"
+        // lib2git needs an empty folder to store the repo clones
+        for a in [ pathOld; pathNew ] do
+            if Directory.Exists(a) then
+                Directory.Delete(a, true)
 
-        // -l [git commit hash] [git commit hash] [OutputPath]
-        elif flag.Contains("-l") then
-            Console.WriteLine("***** git commit")
+        checkoutCommit commit1 location pathOld
+        checkoutCommit commit2 location pathNew
 
-            if argvList.Length <> 4 then
-                failwith "usage: program -l OLD_COMMIT NEW_COMMIT OUTPUTFOLDER"
+        Utils.log $"{pathOld}, {pathNew}, {path}/Output"
 
-            // location of repo
-            let location = Directory.GetCurrentDirectory()
 
-            let commit1 = argvList.Item(1)
-            let commit2 = argvList.Item(2)
+    let runLocal (local: LocalOptions) : unit =
 
-            let path = argvList.Item(3)
-            let pathOld = $"{path}/Old"
-            let pathNew = $"{path}/New"
+        Utils.log "***** local project comparison"
+        Utils.log $"{local.oldFolder}, {local.newFolder}, {local.outputFolder}"
 
-            // lib2git needs an empty folder to store the repo clones
-            for a in [ pathOld; pathNew ] do
-                if Directory.Exists(a) then
-                    Directory.Delete(a, true)
+    let runRemote (remote: RemoteOptions) : unit =
+        Utils.log "***** remote github"
 
-            checkoutCommit commit1 location pathOld
-            checkoutCommit commit2 location pathNew
+        let path = remote.outputFolder
+        let pathOld = $"{path}/Old"
+        let pathNew = $"{path}/New"
+        let url = remote.url
+        let commit1 = remote.oldHash
+        let commit2 = remote.newHash
 
+        for a in [ pathOld; pathNew ] do
+            if Directory.Exists(a) then
+                Directory.Delete(a, true)
```

Comment:
I think we should have a `--force` option, and if the directory exists and `--force` is not provided we should exit with an error message that the directory exists

Reply:
I understand now, fixed